Controllers/WishListsController.cs

```diff
@@ -0,0 +1,256 @@
+            wishlistId = wishList.ContentItem.Id;
+            //add product to wishlist
+            if (productid > 0) {
+                var productPart = _contentManager.Get<ProductPart>(productid);
+                var productattributes = ParseProductAttributes();
+
+                _wishListServices.AddProductToWishList(user, wishList, productPart, productattributes);
+            }
+
+            return RedirectToAction("Index", new { id = wishlistId });
+        }
+
+        [HttpPost]
+        public ActionResult AddToWishList(int wishListid, int productid) {
+            var user = _wca.GetContext().CurrentUser;
+            if (user == null) {
+                return new HttpUnauthorizedResult();
+            }
+            //get selected wishlist
+            var wishList = _wishListServices.GetWishList(user, wishListid);
+
+            if (productid > 0) {
+                var productPart = _contentManager.Get<ProductPart>(productid);
+                var productattributes = ParseProductAttributes();
+
+                _wishListServices.AddProductToWishList(user, wishList, productPart, productattributes);
+            }
+
+            return RedirectToAction("Index", new { id = wishList.ContentItem.Id });
+        }
+
+        [HttpPost]
+        public ActionResult AddToCart(int productId, int quantity = 1) {
+
+            //read attributes from the form
+            var form = HttpContext.Request.Form;
+            var productattributes = form.AllKeys
+                .Where(key => key.StartsWith("attributeKey"))
+                .Select(key => int.Parse(form[key]))
+                .ToDictionary(
+                    key => key, //id of the attribute
+                    key => { //ProductAttributeValueExtended
+                        return new ProductAttributeValueExtended {
+                            Value = form["value_" + key],
+                            ExtendedValue = form["ExtendedValue_" + key],
+                            ExtensionProvider = form["ExtensionProvider_" + key]
+                        };
+                    }
+                );
+
+            _shoppingCart.Add(productId, quantity, productattributes);
+
+            _workflowManager.TriggerEvent("CartUpdated",
+                _wca.GetContext().CurrentSite,
+                () => new Dictionary<string, object> {
+                    {"Cart", _shoppingCart}
+                });
+
+            return RedirectToAction("Index", new { controller = "ShoppingCart" });
+        }
+
+        [HttpPost]
+        public ActionResult RemoveFromWishList(int wishListid, int elementId) {
+            var user = _wca.GetContext().CurrentUser;
+            if (user == null) {
+                return new HttpUnauthorizedResult();
+            }
+            //get selected wishlist
+            var wishList = _wishListServices.GetWishList(user, wishListid);
+
+            if (wishList.ContentItem.Id == wishListid) { //GetWIshList may return the default wish list
```

Comment:
I added a `TryGetWishList` method that returns true if the wishlist is found, false if it had to fallback on the default.